### PR TITLE
Fix US and CA account number padding

### PR DESCRIPTION
**Pull request by JoeSouthan**
We've noticed that we've been seeing some issues with account numbers in PAD; we're padding out too many digits.

This change 
- Adds padding with `_` rather than 0s for bank details within the range of account numbers with 7 to 12 digits - removed by the splitter
- Adds range support for validating account number length
- Fixes `_` from appearing in `iban.account_number`

### CODE REVIEW

File: lib/ibandit/local_details_cleaner.rb [hunk 1]
```diff
@@ -330,9 +330,7 @@ def self.clean_hr_details(local_details)
     def self.clean_hu_details(local_details)
       # This method supports being passed the component IBAN parts, as defined
       # by SWIFT, or a single 16 or 24 digit string.
-      if local_details[:bank_code] || local_details[:branch_code]
-        return local_details
-      end
+      return local_details if local_details[:bank_code] || local_details[:branch_code]
```
Comment (danielroseman) [inline lib/ibandit/local_details_cleaner.rb:333]: Was this change required by rubocop? I think it's much less clear than the original, I would need to remember the precedence of `if` vs `||` to understand what is returned

File: data/structures.yml [hunk 2]
```diff
@@ -1015,7 +1018,10 @@ NZ:
 US:
   :bank_code_length: 9
   :branch_code_length: 0
-  :account_number_length: 17
+  :account_number_length: !ruby/range
+    begin: 1
```
Comment (danielroseman) [inline data/structures.yml:1022]: Is it really possible to have a one-digit account number?

**Review by danielroseman — COMMENTED** [review]
Do we need to update the Canada section of the readme as well?

File: data/structures.yml (hunk 2 above)
Comment (JoeSouthan) [reply data/structures.yml:1022]: I'm not too sure, I'd hope not!

File: lib/ibandit/local_details_cleaner.rb (hunk 1 above)
Comment (JoeSouthan) [reply lib/ibandit/local_details_cleaner.rb:333]: I agree, tried to put brackets around the `||` but Rubocop wasn't having it

**Review by danielroseman — APPROVED** [review]
